thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java

```diff
@@ -236,15 +236,19 @@ public Response evaluate(
     final long safeEndTime = safeEndTime(request.getEnd().getTime());
     request.setEnd(new Date(safeEndTime));
 
-    final AlertApi alert = request.getAlert();
-    ensureExists(alert)
-        .setOwner(new UserApi()
-            .setPrincipal(principal.getName()));
-    // see TE-1172 - ensure the dto passed to the authManager is complete, not only a reference {id:1234}
-    final AlertDTO alertDTO = optional(alert.getId()).map(this::get).orElseGet(() -> toDto(alert));
-    authorizationManager.ensureCanEvaluate(principal, alertDTO);
-
-    final AlertEvaluationApi results = alertEvaluator.evaluate(request);
+    final AlertApi alertApi = ensureExists(request.getAlert())
+        .setOwner(new UserApi().setPrincipal(principal.getName()));
+
+    AlertEvaluationApi results;
+    if (alertApi.getId() != null) {
```

Comment:
This introduces execution complexity. I'd be interested to know what is the longer term motivation for this change. 

My understanding was that we just need to decide on how the auth check is being performed.

Reply:
I was trying to be more explicit about what is considered _new alert eval_ vs _existing alert eval_.
The code that checks for the alert id and loads it from the db is fairly subtle.

Reply:
I don't feel too strongly about this refactor, tho. I'm fine with only checking for the alert id and no other changes.